### PR TITLE
gh-127060: Disable traceback colors in IDLE

### PR DESCRIPTION
**Pull request by vstinner**
Set TERM environment variable to "dumb" to disable traceback colors in IDLE, since IDLE doesn't understand ANSI escape sequences.

* Issue: gh-127060

### CODE REVIEW

File: Lib/idlelib/pyshell.py [hunk 1]
```diff
@@ -1596,6 +1596,9 @@ def main():
     enable_edit = enable_edit or edit_start
     enable_shell = enable_shell or not enable_edit
 
+    # gh-127060: Disable traceback colors
+    os.environ['TERM'] = 'dumb'
```
Comment (serhiy-storchaka) [inline Lib/idlelib/pyshell.py]: Should not it be set in a subprocess that runs user code?

Comment (vstinner) [reply Lib/idlelib/pyshell.py]: os.environ is inherited, but ok, I modified my PR to only set TERM env var in the subprocess.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by vstinner** [comment]
@terryjreedy: Are you ok with this approach for now?

File: Lib/idlelib/pyshell.py [hunk 2]
```diff
@@ -424,7 +424,9 @@ def __init__(self, tkconsole):
     def spawn_subprocess(self):
         if self.subprocess_arglist is None:
             self.subprocess_arglist = self.build_subprocess_arglist()
-        self.rpcsubproc = subprocess.Popen(self.subprocess_arglist)
+        # gh-127060: Disable traceback colors
+        env = dict(os.environ, TERM='dumb')
```
Comment (terryjreedy) [inline Lib/idlelib/pyshell.py:428]:
```suggestion
        env = dict(os.environ, TERM='dumb')  # TODO move this and arglist to __init__.
```

**Review by terryjreedy — APPROVED** [review]
I tested this on macOS and colors were suppressed even when excepthook was set.  I would have been OK with setting in IDLE processes, but being explicit here is fine too.

I believe sys.subprocess_arglist and sys.env could be set in __init__ but will defer this and other cleanups to another PR sometime.

Tests were hanging incomplete, so updated to trigger a proper run.  Will merge if finish before sleep.

**Comment by vstinner** [comment]
Thanks for the review and the merge.

**Pull request merged**